unime/src-tauri/tests/tests/search_query.rs

```diff
@@ -12,95 +12,75 @@ async fn test_credential_search_query() {
-    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_search_sort_query.json");
+    let action = json_example::<Action>("tests/fixtures/actions/credential_add_recent_search.json");
+    let expected_state = json_example::<AppState>("tests/fixtures/states/two_credentials_recent_search_no_current.json");
     assert_state_update(
-        AppStateContainer(Mutex::new(state)),
+        AppStateContainer(Mutex::new(state.clone())),
         vec![action],
         vec![Some(expected_state)],
     )
     .await;
 }
 
+// To check wether searching the same search_hit isn't registered twice.
 #[tokio::test]
 #[serial_test::serial]
-async fn test_connections_search_query() {
+async fn test_credential_add_recent_search_twice() {
     setup_state_file();
     setup_stronghold();
 
-    let state = json_example::<AppState>("tests/fixtures/states/three_connections.json");
-    let action = json_example::<Action>("tests/fixtures/actions/connection_search.json");
-    let expected_state = json_example::<AppState>("tests/fixtures/states/three_connections_search.json");
-    assert_state_update(
-        AppStateContainer(Mutex::new(state)),
-        vec![action],
-        vec![Some(expected_state)],
-    )
-    .await;
-}
-
-#[tokio::test]
-#[serial_test::serial]
-async fn test_connections_sort_query() {
-    setup_state_file();
-    setup_stronghold();
+    let state = json_example::<AppState>("tests/fixtures/states/two_credentials_recent_search.json");
+    let action = json_example::<Action>("tests/fixtures/actions/credential_add_recent_search.json");
 
-    let state = json_example::<AppState>("tests/fixtures/states/three_connections.json");
-    let action = json_example::<Action>("tests/fixtures/actions/connection_sort.json");
-    let expected_state = json_example::<AppState>("tests/fixtures/states/three_connections_sort.json");
     assert_state_update(
-        AppStateContainer(Mutex::new(state)),
+        AppStateContainer(Mutex::new(state.clone())),
         vec![action],
-        vec![Some(expected_state)],
+        vec![Some(state)],
     )
     .await;
 }
 
 #[tokio::test]
 #[serial_test::serial]
-async fn test_connections_search_sort_query() {
+async fn test_credential_delete_recent() {
     setup_state_file();
     setup_stronghold();
 
-    let state = json_example::<AppState>("tests/fixtures/states/three_connections.json");
-    let action = json_example::<Action>("tests/fixtures/actions/connection_search_sort.json");
-    let expected_state = json_example::<AppState>("tests/fixtures/states/three_connections_search_sort.json");
+    let state = json_example::<AppState>("tests/fixtures/states/two_credentials_search_query.json");
+    let action = json_example::<Action>("tests/fixtures/actions/credential_search_delete_recent.json");
+    let expected_state_2 = json_example::<AppState>("tests/fixtures/states/two_credentials_search_delete_recent.json");
```

Comment:
No need to update the name from `expected_state` to `expected_state_2` since the original `expected_state` is removed, right?